training-react/src/app/components/Game/index.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react';
+
+import Board from '../Board';
+
+import calculateWinner from './utils';
+import styles from './styles.scss';
+
+class Game extends Component {
+  state = {
+    history: [{
+      squares: Array(9).fill(null),
+    }],
+    xIsNext: true,
+    stepNumber: 0,
+    isWinner: false
+  };
+
+  getStatus = winner => {
+    let status;
+    if (winner) {
+      status = `Winner: ${this.state.xIsNext ? 'O' : 'X'}`;
+    } else {
+      status = `Next player: ${this.state.xIsNext ? 'X' : 'O'}`;
+    }
+    return status;
+  };
+
+  handleClick = i => {
+    if (!this.props.isWinner) {
+      const history = this.state.history.slice(0, this.state.stepNumber + 1);
+      const current = history[history.length - 1];
+      const squares = [...current.squares];
+      if (calculateWinner(squares) || squares[i]) {
```

Comment:
As you already know there is no winner, i think it should be enough to check if the square is filled.
What if we just check `if (!squares[i])` and set the winner inside setState?
Let me know